### PR TITLE
Feature/text with media

### PR DESCRIPTION
**Pull request by shubanshk**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #<gh-issue-id>

Test URLs:
- Before: https://main--{repo}--{owner}.hlx.live/
- After: https://feature-text-with-media--metafox-sites--bmw-importer.hlx.page/test-pages/index-copy/custom-text/text-with-media

### CODE REVIEW

**Review by Harish-DV — APPROVED** [review]

File: component-models.json [hunk 1]
```diff
@@ -433,6 +433,372 @@
       }
     ]
   },
+  {
+    "id": "text-with-image",
```
[the remark this answers is not in the document]
Comment (shubanshk) [reply component-models.json:437]: Sure, I will group the fields as per the request. Reg Headline Type and Button Styling, they need to have only one option for the dropdown with the field set as hidden. Thus keeping it as it is.

File: component-models.json [hunk 2]
```diff
@@ -433,6 +433,372 @@
...
+          {
+            "name": "Link Button",
+            "value": "linkbutton"
+          }
+        ]
+      }
+    ]
+  },
+  {
+    "id": "text-with-video",
```
[the remark this answers is not in the document]
Comment (shubanshk) [reply component-models.json:587]: Sure, I will group the fields as per the request.

File: component-models.json [hunk 3]
```diff
@@ -433,6 +433,372 @@
...
+        "description":"Defines the text that will appear above the headline."
+      },
+      {
+        "component": "select",
+        "valueType": "string",
+        "name": "style",
+        "value": "",
+        "label": "Eyebrow Type",
+        "description":"Defines the styling of the text that will appear above the headline.",
+        "options": [
```
[the remark this answers is not in the document]
Comment (shubanshk) [reply component-models.json:628]: The ask is to have these selections separately and not using multi-select. Thus keeping as it is

File: component-models.json [hunk 4]
```diff
@@ -433,6 +433,372 @@
...
+        "description": "Defines the text that will appear in the paragraph styling.",
+        "valueType": "string"
+      },
+      {
+        "component": "tab",
+        "label": "Video",
+        "name": "textWithVideoTab2"
+      }, 
+      {
+        "component": "text",
```
[the remark this answers is not in the document]
Comment (shubanshk) [reply component-models.json:680]: Sure, I will group the fields as per the request.

**Comment by shubanshk** [comment]
The test url is https://feature-text-with-media--metafox-sites--bmw-importer.hlx.page/test-pages/index-copy/custom-text/text-with-media

**Review by bharath-yathiraj — APPROVED** [review]
approved

**Pull request merged**